crates/feldera-types/src/serde_with_context/deserialize.rs

```diff
@@ -245,6 +245,61 @@ where
     }
 }
 
+impl<'de, C, K, V> DeserializeWithContext<'de, C> for Arc<BTreeMap<K, V>>
```

Comment:
This shouldn't be necessary. It should be possible to deserialize into `Map<>` and then wrap the result into `Arc`. In fact something like this can work for any type wrapped in `Arc`.

```rust
impl<'de, T, C> DeserializeWithcontext<'de, C> for Arc<T>
where
    T: DeserializeWithContext<'de, C>,
{
    fn deserialize_with_context<D>(deserializer: D, context: &'de C) -> Result<Self, D::Error>
    where
        D: Deserializer<'de>,
    {
        let val = T::deserialized_with_context(deserializer, context)?;
        Ok(Arc::new(val))
    }
}

``